ui/ducks/ens.js

```diff
@@ -177,9 +187,25 @@ export function lookupEnsName(ensName) {
       let address;
       let error;
       try {
```

Comment:
seems like another viable approach here is 
```
const isProviderActive = provider.provider.writable;
        if (!isProviderActive) {
          await dispatch(initializeEnsSlice());
        }
        address = await provider.resolveName(trimmedEnsName);
```

the `writable` property on the `provider` object on the `web3provider` will flip to false when stale/unresponsive, so we don't need to go through the process of attempting to resolve a name and seeing if the attempt times out and then re-initializing. This approach is more streamlined.

Reply:
I've adopted this approach [here](https://github.com/MetaMask/metamask-extension/pull/16335/commits/3f95f2e5c72db98f57d7521a598428f83d055cee)